socks/socks.go

```diff
@@ -162,30 +159,33 @@ func ParseAddr(s string) Addr {
 }
 
 // Handshake fast-tracks SOCKS initialization to get target address to connect.
+// Read RFC 1928 for request and reply structure and sizes.
 func Handshake(rw io.ReadWriter) (Addr, error) {
-	// Read RFC 1928 section 4 for request and reply structure and sizes
-	buf := make([]byte, MaxReqLen)
-
-	_, err := rw.Read(buf) // SOCKS version and auth methods
-	if err != nil {
+	buf := make([]byte, 0xff)
```

Comment:
If I reuse the ReadAddr I will allocate memory twice. Once here for 255 which is the maximum
 of byte, Once inside the ReadAddr function for MaxAddrLen. Is that OK?
Or I can pass the buffer into ReadAddr, which will change the ReadAddr function.